src/vs/workbench/browser/dnd.ts

```diff
@@ -220,18 +220,23 @@ export function fillEditorsDragData(accessor: ServicesAccessor, resourcesOrEdito
 	}));
 
 	const fileSystemResources = resources.filter(({ resource }) => fileService.hasProvider(resource));
+
+	// Separate directories from files
+	const draggedDirectories: IResourceStat[] = fileSystemResources.filter(({ isDirectory }) => isDirectory);
+	const draggedFiles: IResourceStat[] = fileSystemResources.filter(({ isDirectory }) => !isDirectory);
+
 	if (!options?.disableStandardTransfer) {
 
 		// Text: allows to paste into text-capable areas
 		const lineDelimiter = isWindows ? '\r\n' : '\n';
-		event.dataTransfer.setData(DataTransfers.TEXT, fileSystemResources.map(({ resource }) => labelService.getUriLabel(resource, { noPrefix: true })).join(lineDelimiter));
```

Comment:
We want to write both files and dirs in this case

It looks like maybe a few too many lines got touch in this change. Maybe instead of adding `draggedDirectories` and `draggedFiles`, you can make the fix clearer by focusing on the part below where we built the `url-list`. You can use use `fileSystemResources` to get the directories and combine those with the dragged editors. Something like:

```ts
if (draggedEditors.length) {
   event.dataTransfer.setData(CodeDataTransfers.EDITORS, stringify(draggedEditors));
}

const allDraggedResources = [
    ...draggedEditors.map(editor => resource),
    ...fileSystemResources.filter(({ isDirectory }) => isDirectory).map(dir => dir.resource)
]
if (allDraggedResources.length) {
    // set url-list
}
```